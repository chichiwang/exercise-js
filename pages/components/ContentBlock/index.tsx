import React from 'react';

import styles from './styles.module.scss';

interface ContentBlockProps extends React.PropsWithChildren {
}

function ContentBlock({ children }: ContentBlockProps) {
  return (
    <div className={styles.contentBlock}>
      {children}
    </div>
  );
}

export default ContentBlock;
