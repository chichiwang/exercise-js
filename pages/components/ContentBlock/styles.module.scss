@use "sass:map";
@use "app/var.module";

.contentBlock {
  margin: 0 auto;
  max-width: map.get(var.$content, 'article-max-width');
}
