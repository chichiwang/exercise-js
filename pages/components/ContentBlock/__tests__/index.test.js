import React from 'react';
import renderer from 'react-test-renderer';

import ContentBlock from '../index';

describe('ContentBlock Component', () => {
  test('matches no-child snapshot', () => {
    const tree = renderer.create(
      <ContentBlock />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('matches child snapshot', () => {
    const tree = renderer.create(
      <ContentBlock>
        <p>Block content</p>
      </ContentBlock>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
