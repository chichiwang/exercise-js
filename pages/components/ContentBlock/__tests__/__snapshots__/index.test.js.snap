// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentBlock Component matches child snapshot 1`] = `
<div
  className="contentBlock"
>
  <p>
    Block content
  </p>
</div>
`;

exports[`ContentBlock Component matches no-child snapshot 1`] = `
<div
  className="contentBlock"
/>
`;
